shared/js/trackers.js

```diff
@@ -233,6 +237,17 @@ require.scopes.trackers = (function() {
         return match
     }
 
+    function checkSurrogateList(url) {
+        const b64dataheader = 'data:application/javascript;base64,'
+
+        for (item of Object.keys(surrogateList)) {
+            if (surrogateList[item].regex.test(url)) {
+                return {redirectUrl: b64dataheader + surrogateList[item].snippet}
```

Comment:
So this is all being done locally instead of redirecting to another endpoint it seems? Sorry if I missed it, but why the deviation?

Reply:
I think this was always the plan.

redirecting with an endpoint would simplify the client but I think we're talking about redirecting URLs from 75% of the web, which has negative privacy implications. I might be wrong but also aren't there cross domain issues with HTTPS? This way is also more transparent since we send resources.txt unchanged.

Reply:
That's a good point for both CSP and CORS. I think we could always make a privacy implication out of something, but in this particular case, I think that's a witch hunt. All that said, the CORS and CSP constrains seem non trivial and per server configurable, which this could avoid.

I think this is, as programmed, way too brittle to be run on, as you say, 75% of the web. This stuff needs to be bullet proof IMO.

Reply:
Am also wondering this^^. Perhaps its just the optics.